File: spring-boot-java-web/src/main/java/com/example/webjsp/controller/CityController.java
package com.example.webjsp.controller;

import com.example.webjsp.service.CityService;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;

import java.util.HashMap;
import java.util.List;

@Slf4j
@Controller
public class CityController {

    private final CityService cityService;
    public CityController(@Autowired CityService cityService) {
        this.cityService = cityService;
    }

    @GetMapping("/city/list")
    public String list(Model model) {
        List<HashMap<String, Object>> items = cityService.selectAll();
        model.addAttribute("cityItems", items);
        log.debug("/city/list items: {}",items);
        return "city/list";
    }
}

File: spring-boot-kotlin-web/src/main/kotlin/com/example/webjsp/controller/CityController.kt
package com.example.webjsp.controller

import com.example.webjsp.service.CityService
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping

@Controller
@RequestMapping("/city")
class CityController {

  @Autowired
  lateinit var cityService: CityService

  @GetMapping("/list")
  fun list(model: Model): String {
    val items = cityService.selectAll()
    model.addAttribute("cityItems", items)
    log.trace("/city/list items: {}", items)
    return "city/list"
  }

  companion object {
    val log: Logger = LoggerFactory.getLogger(CityController::class.java)
  }
}
